app/src/main/java/com/lagradost/cloudstream3/ui/download/DownloadFragment.kt

```diff
@@ -280,4 +298,118 @@ class DownloadFragment : Fragment() {
 
         fixPaddingStatusbar(binding?.downloadRoot)
     }
+}
+
+class SwipeToDeleteCallback(private val adapter: DownloadHeaderAdapter) : ItemTouchHelper.Callback() {
+
+    private var deleteInitiated = false
+
+    override fun getMovementFlags(
+        recyclerView: RecyclerView,
+        viewHolder: RecyclerView.ViewHolder
+    ): Int {
+        val position = viewHolder.bindingAdapterPosition
+        val item = adapter.cardList[position]
+        if (item.data.type.isEpisodeBased()) return 0
+        return makeMovementFlags(ItemTouchHelper.LEFT, 0)
+    }
+
+    override fun isLongPressDragEnabled(): Boolean = true
+
+    override fun isItemViewSwipeEnabled(): Boolean = false
+
+    override fun onMove(
+        recyclerView: RecyclerView,
+        viewHolder: RecyclerView.ViewHolder,
+        target: RecyclerView.ViewHolder
+    ): Boolean = false
+
+    override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {}
+
+    private fun handleDelete(viewHolder: RecyclerView.ViewHolder) {
+        if (deleteInitiated) return
+
+        deleteInitiated = true
+        val position = viewHolder.bindingAdapterPosition
+        val item = adapter.cardList[position]
+
+        CoroutineScope(Dispatchers.Main).launch {
```

Comment:
I guess we should probably use `runOnMainThread` here.

This is just a reminder for me.